### PR TITLE
Add sensible scalac options

### PR DESCRIPTION
**Pull request by mdedetrich**
Self explanatory

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -104,3 +104,20 @@ ThisBuild / githubWorkflowOSes := Seq("ubuntu-latest", "macos-latest", "windows-
...
+ThisBuild / scalacOptions ++= List(
+  "-unchecked",
+  "-deprecation",
+  "-language:_",
+  "-encoding",
+  "UTF-8")
+
+ThisBuild / scalacOptions ++= {
+  if (insideCI.value) {
+    val log = sLog.value
```
Comment (pjfanning) [inline build.sbt:117]: how about when a human needs to release this?

Comment (mdedetrich) [reply build.sbt:117]: Forgot about this, will be handled by our sbt plugin. Removing now

Comment (mdedetrich) [reply build.sbt]: @pjfanning Fixed

**Review by pjfanning — APPROVED** [review]
lgtm

**Pull request merged**